### DISCUSSION TITLE
Handle incremental data loading keeps switching between two layers

### DISCUSSION THREAD

**Discussion by acebelowzero** [visgl/deck.gl#5243] [Q&A]
When I start my method to begin loading all the data I have two scenegraph layers in my layers but they keep replacing each other.

```
async *filterByDate() {
      let iter = 0;
      let offset = 0;
      ///this.loading = true;
      for (let it = 0; it < 10; it++) {
         const chunk = await axios
          .post("/api/DronesbyDate", {
            To: this.dateTo,
            From: this.dateFrom,
            offset: offset,
          });
          offset += 5000;
          yield chunk.data;
      }
    },
```
```
upd() {
            const layers = [
            new ScenegraphLayer({
              id: "scenegraph-layer",
              data: this.filterByDate(),
              pickable: true,
              scenegraph: MODEL_URL,
              sizeScale: 10,
              getOrientation: (d) => [0, d.Yaw, 90],
              getPosition: (d) => [d.Longitude, d.Latitude, d.Altitude],
              getColor: (d) => [238, 104, 103],
            }),
            new ScenegraphLayer({
              id: "home",
              data: this.filterByDate(),
              pickable: true,
              scenegraph: MODEL_URL,
              sizeScale: 10,
              getOrientation: (d) => [0, 0, 90],
              getPosition: (d) => [d.HomeLongitude, d.HomeLatitude, d.AltitudeHeight],
              getColor: (d) => [123, 67, 122]
            }),
```
Shouldnt the layers build up together instead of replacing each other over and over?

**Comment by Pessimistress** [comment]
Not sure what you mean by “ they keep replacing each other.” Note that every time `upd()` is called you are restarting the fetching by creating a new async generator.

**Comment by acebelowzero** [comment]
I called udp once and it makes once fetch and ill have this data
![layer one](https://user-images.githubusercontent.com/64754588/101658884-afe07300-3a02-11eb-9c34-82758dbfb46e.PNG)

then it will switch to my other layer.
![layertwo](https://user-images.githubusercontent.com/64754588/101659142-f635d200-3a02-11eb-9a04-150a9eba7cd4.PNG)

It will keep doing this over and over till the loop has stopped.
But when I use column layer I have no problem. both layers in the layer array load together as they should but only when I use scenegraph layer is when they replace each other. @Pessimistress

**Comment by Pessimistress** [comment]
I cannot reproduce this issue: https://codepen.io/Pessimistress/pen/oNNdzMd

I suggest you:
- Check that you are using the latest release
- Create a Codepen/CodeSandbox that reproduces this issue, and open a bug report
